provider/spark.go

```diff
@@ -1119,18 +1177,21 @@ func (s *S3Materialization) ID() MaterializationID {
 }
 
 func (s *S3Materialization) NumRows() (int64, error) {
-	numRows, err := s.store.ResourceRowCt(s.Key)
+	numRows, err := s.store.ResourceRowCt(s.id)
 	if err != nil {
 		return 0, err
 	}
 	return int64(numRows), nil
 }
 
 func (s *S3Materialization) IterateSegment(begin, end int64) (FeatureIterator, error) {
-	stream, err := s.store.ResourceStreamConv(s.Key, begin)
+	stream, err := s.store.ResourceMultiPartStream(s.id)
 	if err != nil {
 		return nil, err
 	}
+	for i := int64(0); i < begin; i++ {
```

Comment:
Wasn't there a function that does this?

Reply:
Added an implementation for multiple files